### PR TITLE
[WFLY-19130] Publish Wildfly channels

### PR DESCRIPTION
**Pull request by spyrkob**
Issue: https://issues.redhat.com/browse/WFLY-19130

### CODE REVIEW

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+|===
+| Manifest GA
+| Artifacts
+| Depends on
+
+|`org.wildfly.channels:wildfly-ee`
+|wildfly-ee-galleon-pack and its dependencies
+| -
+
+|`org.wildfly.channels:wildfly`
```
Comment (spyrkob) [inline build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:100]: I'm wondering if a better name would be something like `org.wildfly.channels:wildfly-full`

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+WildFly project currently creates three feature packs: `wildfly-ee-galleon-pack`, `wildfly-galleon-pack` and `preview-galleon-pack`.
```
Comment (bstansberry) [inline build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:100]: These are 'wildfly-ee', 'wildfly' and 'wildfly-preview'. The unfortunately varied artifact ids are not how we refer to them in documentation unless we are specifically documenting artifactIds. (Also, the artifactId of wildfly-preview is 'wildfly-preview-feature-pack'.)

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc (hunk 1 above)
Comment (bstansberry) [reply build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:124]: That would be a bummer.

If you to provision what almost everyone regards as WildFly you use the 'wildfly' feature pack. Adding "full" is confusing.

The channels are a bit different from the feature packs though; using the 'wildfly' FP means you unavoidably get 'wildfly-ee' with no need for the user to even be aware that's happened.

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+| Manifest GA
+| Artifacts
+| Depends on
+
+|`org.wildfly.channels:wildfly-ee`
+|wildfly-ee-galleon-pack and its dependencies
+| -
+
+|`org.wildfly.channels:wildfly`
+| wildfly-galleon-pack and its dependencies
```
Comment (bstansberry) [inline build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:125]: 'wildfly' feature pack and its dependencies

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+|`org.wildfly.channels:wildfly-ee`
+
+|`org.wildfly.channels:wildfly-preview`
+| wildfly-preview-galleon-pack and its dependencies
```
Comment (bstansberry) [inline build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:129]: 'wildfly-preview' feature pack and its dependencies

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+|===
+
+==== Manifest metadata Requirements
+
+The manifest file defines a few metadata fields:
+* name - human readable name of the manifest. Should contain the version of the server.
+* id - identifier of the manifest used to resolve manifest dependencies. Should be the same as the GA of the manifest.
+* description - human readable description of the manifest.
+* requires - id and optionally maven coordinates of dependent manifests.
```
Comment (bstansberry) [inline build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:140]: s/dependent/dependency

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+=== Hard Requirements
+
+* WildFly manifests are published in Maven repository during release.
+* Provisioning tools using wildfly-channels can use the published manifest to provision and update WildFly server.
+* Different WildFly features packs created in the project are able to use different versions of the same artifacts.
+* The versions of the manifests correspond to Wildfly releases. When using version `32.0.0.Final`, the user should
```
Comment (bstansberry) [inline build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:147]: s/Wildfly/WildFly

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+* description - human readable description of the manifest.
+* requires - id and optionally maven coordinates of dependent manifests.
+
+=== Hard Requirements
+
+* WildFly manifests are published in Maven repository during release.
+* Provisioning tools using wildfly-channels can use the published manifest to provision and update WildFly server.
+* Different WildFly features packs created in the project are able to use different versions of the same artifacts.
+* The versions of the manifests correspond to Wildfly releases. When using version `32.0.0.Final`, the user should 
+receive manifest for Wildfly 32.0.0 Final
```
Comment (bstansberry) [inline build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:148]: s/Wildfly/WildFly

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+N/A
+
+== Test Plan
+
+* Existing WildFly tests already use the channels
+* Add test in Prospero to provision the latest WildFly version
+
+== Community Documentation
+
+Update Prospero documentation and examples with provisioning WildFly server using published manifests.
```
Comment (bstansberry) [inline build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:191]: We should update https://docs.wildfly.org/wildfly-maven-plugin/releases/4.2/provision-example.html as well.

File: build/WFLY-19130_publish_Wildfly_channel_manifest.adoc (hunk 1 above)
Comment (spyrkob) [reply build/WFLY-19130_publish_Wildfly_channel_manifest.adoc:124]: It's mostly for consistency - the other two channels have a name of `wildfly-*`, so it feels like channel name `wildfly` misses something :)
On the other hand, like you said, `wildfly` is consistent with naming of the feature packs.

**Comment by spyrkob** [comment]
@bstansberry do we want to think about adding the provisioning metadata to the WildFly zip we distribute? Adding `.galleon/provisioning.xml`, `.installation/installer-channels.yaml` and `.installation/manifest.yaml` would make it ready to be used with Prospero to e.g. install new feature-packs

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
@spyrkob I've approved this as I think it's fine for WF 32 Beta.

Re adding more metadata to the standard zips, what would that require in terms of build process? We provide a .galleon/provisioning.xml, but just an empty .installation dir. Adding more makes sense and is something we can add into this analysis for 32 Final, if the mechanics involved in producing it work out.

**Comment by darranl** [comment]
@jamezp As a member of the feature team are you able to review / approve this one please?

**Review by jamezp — APPROVED** [review]
My apologies. I approved the WildFly PR and forgot about this one. :)

**Pull request merged**